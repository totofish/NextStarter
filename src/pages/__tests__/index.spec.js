import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import App from '../index';

const url = {
  pathname: '/',
  query: {},
  back: jest.fn(),
  push: jest.fn(),
  pushTo: jest.fn(),
  replace: jest.fn(),
  replaceTo: jest.fn(),
};
const app = mount((
  <App
    stars={1999}
    url={url}
    info={{
      message: '127.0.0.1',
      type: 'IP'
    }}
    ip="127.0.0.1"
  />
));

describe('Index With Enzyme', () => {
  it('Have text "127.0.0.1"', () => {
    expect(app.html()).toContain('127.0.0.1');
  });

  it('Have one dom <nav>', () => {
    expect(app.find('nav')).toHaveLength(1);
  });
});

describe('Index Snapshot Testing', () => {
  it('Index Dom', () => {
    const component = renderer.create((
      <App
        stars={1999}
        url={url}
        info={{
          message: '127.0.0.1',
          type: 'IP'
        }}
        ip="127.0.0.1"
      />
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
