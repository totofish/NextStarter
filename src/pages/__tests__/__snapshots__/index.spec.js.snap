// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index Snapshot Testing Index Dom 1`] = `
<div
  className="jsx-3955691776 layout"
>
  <header
    className="jsx-3955691776"
  >
    <nav
      className="jsx-3955691776"
    >
      <ul
        className="jsx-3955691776"
      >
        <li
          className="jsx-3955691776 active"
          onClick={[Function]}
        >
          <a
            className="jsx-3955691776"
          >
            Index
          </a>
        </li>
        <li
          className="jsx-3955691776 "
          onClick={[Function]}
        >
          <a
            className="jsx-3955691776"
          >
            page / a
          </a>
        </li>
        <li
          className="jsx-3955691776 "
          onClick={[Function]}
        >
          <a
            className="jsx-3955691776"
          >
            page / b
          </a>
        </li>
        <li
          className="jsx-3955691776 "
          onClick={[Function]}
        >
          <a
            className="jsx-3955691776"
          >
            page / c
          </a>
        </li>
        <li
          className="jsx-3955691776 "
          onClick={[Function]}
        >
          <a
            className="jsx-3955691776"
          >
            page / d
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <i
    className="jsx-3955691776 material-icons"
  >
    
  </i>
  <h1
    className="jsx-3955691776 title"
  >
    Index
  </h1>
  <i
    className="jsx-3955691776 material-icons"
  >
    
  </i>
  <h4
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    NEXT.JS STARS :
  </h4>
  1999
  <h4
    style={
      Object {
        "marginBottom": 10,
      }
    }
  >
    IP :
  </h4>
  127.0.0.1
</div>
`;
